### ISSUE TITLE
Error MT1007: Failed to launch the application. Object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by bradleymarshall** [xamarin/xamarin-macios#3803]
### Steps to Reproduce
1. Launch iOS application in debugger from Visual Studio.

### Expected Behavior
Application launches with debugger attached.

### Actual Behavior
Application terminates with error:

Launch failed. The app 'App.iOS' could not be launched on 'Device'. Error: error MT1007: Failed to launch the application '[debug path]' on the device 'Device': Object reference not set to an instance of an object. You can still launch the application manually by tapping on it.
. Please check the logs for more details.

### Logs Entry
(See comments below)

### Environment
```
Microsoft Visual Studio Professional 2015
Version 14.0.25420.01 Update 3
Microsoft .NET Framework
Version 4.7.02556

Installed Version: Professional

Visual Basic 2015   00322-50350-16355-AA360
Microsoft Visual Basic 2015

Visual C# 2015   00322-50350-16355-AA360
Microsoft Visual C# 2015

Visual C++ 2015   00322-50350-16355-AA360
Microsoft Visual C++ 2015

Application Insights Tools for Visual Studio Package   7.18.00214.2
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2015.1 (Beta8)   14.1.11107.0
ASP.NET and Web Tools 2015.1 (Beta8)

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
For additional information, visit http://www.asp.net/

Common Azure Tools   1.8
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Devart Code Compare   4.2.236
Devart Code Compare
Copyright (c) 2012-2017 Devart. All rights reserved.
http://www.devart.com/codecompare/

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

Merq   1.1.17-rc (cba4571)
Command Bus, Event Stream and Async Manager for Visual Studio extensions.

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Mono Debugging for Visual Studio   4.8.4-pre (3fe64e3)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   3.4.4
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

SQL Server Data Tools   14.0.60519.0
Microsoft SQL Server Data Tools

TypeScript   1.8.34.0
TypeScript tools for Visual Studio

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VisualSVN   5.1.8
Integration with Subversion version control. For more information about VisualSVN, see the VisualSVN website at http://www.visualsvn.com
Copyright © 2005-2017 VisualSVN Ltd. All rights reserved.

Xamarin   4.8.0.760 (fc93f3f5b)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   4.8.192 (4c64462bc)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin.Android   8.1.0.13 (f241f51)
Visual Studio extension to enable development for Xamarin.Android.

Xamarin.Android SDK   8.1.5.0 (HEAD/75f8c6838)
Xamarin.Android Reference Assemblies and MSBuild support.

Xamarin.iOS   11.6.1.4 (db807ec)
Visual Studio extension to enable development for Xamarin.iOS.

Xamarin.iOS and Xamarin.Mac SDK   11.6.1.4 (db807ec)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

**Comment by bradleymarshall** [comment]
My application deploys successfully, and can be launched manually on the device. However, the debugger always crashes with this error at the point of launching the app.

**Comment by bradleymarshall** [comment]
Here's the log excerpt in file form.

[Xamarin-iOS-Bug3803.log](https://github.com/xamarin/xamarin-macios/files/1839155/Xamarin-iOS-Bug3803.log)

**Comment by bradleymarshall** [comment]
**Device:**
iPhone 6 A1586
iOS 11.2.1

**Comment by rolfbjarne** [comment]
I believe this has already been fixed in our next major release, `d15-7`, which is available in the alpha and beta channels. Could you try it and see if it's working or not?

**Comment by mandel-macaque** [comment]
In order to change the update channel you can follow the instructions provided in the link:

https://developer.xamarin.com/recipes/cross-platform/ide/change_updates_channel/

**Comment by bradleymarshall** [comment]
I will try that today and let you know. Thanks!

**Comment by bradleymarshall** [comment]
Looks like this update requires Visual Studio 2017, which I am not currently using. I will explore the option of upgrading, but this will likely take some time.

**Comment by bradleymarshall** [comment]
Issue was resolved by updating my iPhone 6 to iOS 11.2.6.

**Comment by VincentDondain** [comment]
Closing since the issue seems to have been fixed (update/restart). Feel free to reopen this issue if you encounter the same problem again.
